client/src/partials/fiscal/fiscal.openingBalance.js

```diff
@@ -0,0 +1,119 @@
+angular.module('bhima.controllers')
+  .controller('FiscalOpeningBalanceController', FiscalOpeningBalanceController);
+
+FiscalOpeningBalanceController.$inject = [
+  '$state', 'AccountStoreService', 'FiscalService', 'NotifyService', 'util', 'moment'
+];
+
+/**
+ * This controller is responsible for handling the opening balance of the new fiscal year.
+ */
+function FiscalOpeningBalanceController($state, AccountStore, Fiscal, Notify, util, moment) {
+  var vm = this;
+
+  var fiscalYearId = $state.params.id;
+
+  // expose to the view
+  vm.showAccountFilter  = false;
+  vm.editBalanceEnabled = false;
+  vm.showFilter = showFilter;
+  vm.editBalance = editBalance;
+  vm.submit = submit;
+
+  // get fiscal year
+  Fiscal.read(fiscalYearId)
+  .then(function (fy) {
+    vm.fiscal = fy;
+    return fy.previous_fiscal_year_id;
+  })
+  .then(hasPrevious)
+  .catch(Notify.handleError);
+
+  // get account
+  Fiscal.periodicBalance({
+    id: fiscalYearId,
+    period_number: 0
+  })
+  .then(function (list) {
+    vm.accounts = list;
+  })
+  .catch(Notify.handleError);
+
+  /**
+   * @function submit
+   */
+  function submit(form) {
+    vm.balanced = hasBalancedAccount();
+
+    if (!vm.balanced) { return ; }
+
+    // set the fiscal year opening balance
+    Fiscal.setOpeningBalance({
+      id: fiscalYearId,
+      fiscal: vm.fiscal,
+      accounts: vm.accounts
+    })
+    .then(function () {
+      Notify.success(vm.previousFiscalYearExist ? 'FORM.INFO.IMPORT_SUCCESS' : 'FORM.INFO.SAVE_SUCCESS');
+    })
+    .catch(Notify.handleError);
+
+  }
+
+  /**
+   * @function showFilter
+   * @description show a filter for finding an account
+   */
+  function showFilter() {
+    vm.showAccountFilter = !vm.showAccountFilter;
+  }
+
+  /**
+   * @function editBalance
+   * @description edit the opening Balance
+   */
+  function editBalance() {
+    vm.editBalanceEnabled = !vm.editBalanceEnabled;
+  }
+
+  /**
+   * @function hasBalancedAccount
+   * @description check if accounts are balanced
+   */
+  function hasBalancedAccount() {
```

Comment:
This is a nicely written function.

Can balances be negative?  I don't see any code preventing this.